keras_cv/models/segmentation/deeplab_v3_plus/deeplab_v3_plus.py

```diff
@@ -0,0 +1,221 @@
+            backbone to use in many cases is the:
+            `keras_cv.models.ResNet50V2Backbone.from_preset("resnet50_v2_imagenet")`.
+        projection_filters: int, number of filters in the
+            convolution layer projecting low-level features from the `backbone`.
+            The default value is set to `48`, as per the
+            [TensorFlow implementation of DeepLab](https://github.com/tensorflow/models/blob/master/research/deeplab/model.py#L676).
+        spatial_pyramid_pooling: (Optional) a `keras.layers.Layer`. Also known
+            as Atrous Spatial Pyramid Pooling (ASPP). Performs spatial pooling
+            on different spatial levels in the pyramid, with dilation. If
+            provided, the feature map from the backbone is passed to it inside
+            the DeepLabV3 Encoder, otherwise
+            `keras_cv.layers.spatial_pyramid.SpatialPyramidPooling` is used.
+        segmentation_head: (Optional) a `keras.layers.Layer`. If provided, the
+            outputs of the DeepLabV3 encoder is passed to this layer and it
+            should predict the segmentation mask based on feature from backbone
+            and feature from decoder, otherwise a similar architecture is used.
+        dropout (Optional) float, The dropout rate applied to the outputs of the
+            decoder in the `segmentation_head`.
+
+    Examples:
+    ```python
+    import tensorflow as tf
+    import keras_cv
+
+    images = tf.ones(shape=(1, 96, 96, 3))
+    labels = tf.zeros(shape=(1, 96, 96, 1))
+    backbone = keras_cv.models.ResNet50V2Backbone(input_shape=[96, 96, 3])
+    model = keras_cv.models.segmentation.DeepLabV3Plus(
+        num_classes=1, backbone=backbone,
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Train model
+    model.compile(
+        optimizer="adam",
+        loss=keras.losses.BinaryCrossentropy(from_logits=False),
+        metrics=["accuracy"],
+    )
+    model.fit(images, labels, epochs=3)
+    ```
+    """
+
+    def __init__(
+        self,
+        num_classes,
+        backbone,
+        projection_filters=48,
+        spatial_pyramid_pooling=None,
+        segmentation_head=None,
+        dropout=0.0,
+        **kwargs,
+    ):
+        if not isinstance(backbone, keras.layers.Layer) or not isinstance(
+            backbone, keras.Model
+        ):
+            raise ValueError(
+                "Argument `backbone` must be a `keras.layers.Layer` instance "
+                f" or `keras.Model`. Received instead "
+                f"backbone={backbone} (of type {type(backbone)})."
+            )
+
+        inputs = backbone.input
+
+        final_backbone_pyramid_output = backbone.get_layer(
+            list(backbone.pyramid_level_inputs.values())[-1]
+        ).output
+        feature_extractor = keras.Model(
+            inputs=backbone.input,
+            outputs=final_backbone_pyramid_output,
+            name="feature-extractor",
+        )
+        feature_map = feature_extractor(inputs)
```

Comment:
Re-commenting this from earlier: this isn't really a feature map -- this is just one set of features. I think we should name it accordingly (something like `backbone_outputs` or `encoded_features`)